src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1572,6 +1632,43 @@ private IEnumerable<ODataProperty> CreateStructuralPropertyBag(SelectExpandNode
             return properties;
         }
 
+        /// <summary>
+        /// Creates the <see cref="ODataStreamPropertyInfo"/> to be written for the given stream property.
+        /// </summary>
+        /// <param name="structuralProperty">The EDM structural property being written.</param>
+        /// <param name="resourceContext">The context for the entity instance being written.</param>
+        /// <returns>The <see cref="ODataStreamPropertyInfo"/> to write.</returns>
+        public virtual ODataStreamPropertyInfo CreateStreamProperty(IEdmStructuralProperty structuralProperty, ResourceContext resourceContext)
+        {
+            if (structuralProperty == null)
+            {
+                throw Error.ArgumentNull("structuralProperty");
+            }
+
+            if (resourceContext == null)
+            {
+                throw Error.ArgumentNull("resourceContext");
+            }
+
+            if (structuralProperty.Type == null || !structuralProperty.Type.IsStream())
+            {
+                return null;
+            }
+
+            if (resourceContext.SerializerContext.MetadataLevel != ODataMetadataLevel.FullMetadata)
+            {
+                return null;
+            }
```

Comment:
If the stream property does not exactly match convention (i.e., we know the mediatype of the stream) then we should explicitly write the stream property info rather than let ODL generate it with default values.

Reply:
Yes, we'd figure out the solution to let customer set the value. It's next step.

---
In reply to: [574804726](https://github.com/OData/WebApi/pull/2401#discussion_r574804726) [](ancestors = 574804726)